### ISSUE TITLE
Add a GitHub action for publishing the package

### ISSUE THREAD

**Issue by tonycaputome** [Cadienvan/key-value-cache#12]
I think we should automate the publishing process after passing all necessary tests.
We can do it with the github actions.

**Comment by Cadienvan** [comment]
Hi @tonycaputome , if you check the new scaffolding we have something similar, I don't use GitHub actions but I go with npm version and changelog generator paired with commit lint and semver, what do you think? Do you have some spare time to go for it or do you want me to go with that?

**Comment by tonycaputome** [comment]
It's ok having some scripts, the actions just automate these things when you push on particular branch or when you merge a PR.

There are a couple of actions on the marketplace 
https://github.com/marketplace/actions/npm-publish
https://github.com/marketplace/actions/automated-releases-for-npm-packages

This is an example where you can run also manually an action
https://dev.to/astagi/publish-to-npm-using-github-actions-23fn

Basically you can run an action on every event that occurs on a repo.

**Comment by tonycaputome** [comment]
https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs#using-the-nodejs-starter-workflow

**Comment by Cadienvan** [comment]
Thanks Antonio,
I'll check the resources as soon as I can, that's for sure!

Indeed, feel free to do it by yourself if you don't want to wait for me, I could take weeks before looking at it properly.

**Comment by tonycaputome** [comment]
I am busy now, but I'll do a PR if I can

**Comment by Cadienvan** [comment]
Closing to prevent having an open issue for something which isn't, in fact, an issue but a proposal.
